### PR TITLE
Attempt to fix the Kafka flaky tests

### LINKED ISSUE #16371: SslKafkaConsumerTest and SaslKafkaConsumerTest are both flaky
Both fail frequently these days with `java.net.SocketTimeoutException: Read timed out`, e.g: https://github.com/quarkusio/quarkus/pull/16360#issuecomment-815985649

Before the failure, there many warnings like:
```
2021-04-08T16:34:51.7430524Z 2021-04-08 16:34:51,308 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | test-ssl-producer) [Producer clientId=test-ssl-producer] Error while fetching metadata with correlation id 867 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7445567Z 2021-04-08 16:34:51,308 WARN  [org.apa.kaf.cli.NetworkClient] (executor-thread-1) [Consumer clientId=consumer-test-consumer-1, groupId=test-consumer] Error while fetching metadata with correlation id 283 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7461688Z 2021-04-08 16:34:51,411 WARN  [org.apa.kaf.cli.NetworkClient] (kafka-producer-network-thread | test-ssl-producer) [Producer clientId=test-ssl-producer] Error while fetching metadata with correlation id 868 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
2021-04-08T16:34:51.7476286Z 2021-04-08 16:34:51,412 WARN  [org.apa.kaf.cli.NetworkClient] (executor-thread-1) [Consumer clientId=consumer-test-consumer-1, groupId=test-consumer] Error while fetching metadata with correlation id 284 : {test-ssl-consumer=LEADER_NOT_AVAILABLE}
```
Further up in the log:
```
2021-04-08T16:34:51.5863238Z 2021-04-08 16:34:46,520 WARN  [kaf.con.RequestSendThread] (Controller-1-to-broker-1-send-thread) [RequestSendThread controllerId=1] Controller 1's connection to broker fv-az132-838.internal.cloudapp.net:19099 (id: 1 rack: null) was unsuccessful: java.net.SocketTimeoutException: Failed to connect within 30000 ms
2021-04-08T16:34:51.5866133Z 	at kafka.controller.RequestSendThread.brokerReady(ControllerChannelManager.scala:293)
2021-04-08T16:34:51.5871267Z 	at kafka.controller.RequestSendThread.doWork(ControllerChannelManager.scala:246)
2021-04-08T16:34:51.5875378Z 	at kafka.utils.ShutdownableThread.run(ShutdownableThread.scala:96)
```
And this at beginning of the test doesn't look healthy either:
```
2021-04-08T16:33:19.7337164Z 2021-04-08 16:33:13,291 WARN  [kaf.ser.BrokerMetadataCheckpoint] (pool-5-thread-1) No meta.properties file under dir /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-ssl/kafka/broker1/meta.properties
2021-04-08T16:33:19.7340785Z 2021-04-08 16:33:16,231 INFO  [io.deb.kaf.KafkaServer] (pool-5-thread-1) Started Kafka server 1 at localhost:19099 with storage in /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-ssl/kafka/broker1
2021-04-08T16:33:19.7344253Z 2021-04-08 16:33:16,232 INFO  [sta.cha.logger] (controller-event-thread) [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers Set(1) for 0 partitions
2021-04-08T16:33:19.7346875Z 2021-04-08 16:33:16,397 WARN  [kaf.ser.BrokerMetadataCheckpoint] (pool-5-thread-1) No meta.properties file under dir /home/runner/work/quarkus/quarkus/integration-tests/kafka/target/data/kafka-data-sasl/kafka/broker1/meta.properties
2021-04-08T16:33:19.7349939Z 2021-04-08 16:33:16,845 INFO  [sta.cha.logger] (controller-event-thread) [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers Set(1) for 0 partitions
2021-04-08T16:33:19.7354184Z 2021-04-08 16:33:16,854 WARN  [org.apa.kaf.com.uti.AppInfoParser] (pool-5-thread-1) Error registering AppInfo mbean: javax.management.InstanceAlreadyExistsException: kafka.server:type=app-info,id=1
2021-04-08T16:33:19.7358057Z 	at java.management/com.sun.jmx.mbeanserver.Repository.addMBean(Repository.java:436)
2021-04-08T16:33:19.7363800Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerWithRepository(DefaultMBeanServerInterceptor.java:1855)
2021-04-08T16:33:19.7367574Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerDynamicMBean(DefaultMBeanServerInterceptor.java:955)
2021-04-08T16:33:19.7371146Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerObject(DefaultMBeanServerInterceptor.java:890)
2021-04-08T16:33:19.7374303Z 	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.registerMBean(DefaultMBeanServerInterceptor.java:320)
2021-04-08T16:33:19.7378691Z 	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.registerMBean(JmxMBeanServer.java:522)
2021-04-08T16:33:19.7381955Z 	at org.apache.kafka.common.utils.AppInfoParser.registerAppInfo(AppInfoParser.java:64)
2021-04-08T16:33:19.7383571Z 	at kafka.server.KafkaServer.startup(KafkaServer.scala:393)
2021-04-08T16:33:19.7385533Z 	at io.debezium.kafka.KafkaServer.startup(KafkaServer.java:220)
2021-04-08T16:33:19.7387673Z 	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4772)
2021-04-08T16:33:19.7389727Z 	at io.debezium.kafka.KafkaCluster.startup(KafkaCluster.java:250)
2021-04-08T16:33:19.7391603Z 	at io.quarkus.it.kafka.KafkaSASLTestResource.start(KafkaSASLTestResource.java:49)
2021-04-08T16:33:19.7399148Z 	at io.quarkus.test.common.TestResourceManager.lambda$start$1(TestResourceManager.java:107)
2021-04-08T16:33:19.7427944Z 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
2021-04-08T16:33:19.7442008Z 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
2021-04-08T16:33:19.7443548Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-04-08T16:33:19.7446072Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-04-08T16:33:19.7447292Z 	at java.base/java.lang.Thread.run(Thread.java:834)
```

### LINKED ISSUE #16415: KafkaStreamsITCase is flaky causing endless TimeoutException warnings leading to job being aborted
E.g. [here](https://github.com/quarkusio/quarkus/runs/2309475289?check_suite_focus=true) and multiple times in #16249:
```
2021-04-09 21:29:04,547 INFO  [io.quarkus] (main) Profile prod activated. 
2021-04-09 21:29:04,547 INFO  [io.quarkus] (main) Installed features: [cdi, kafka-streams, micrometer, resteasy, resteasy-jackson, smallrye-health]
2021-04-09 21:29:14,537 ERROR [io.qua.kaf.str.run.KafkaStreamsProducer] (pool-4-thread-1) Failed to get topic names from broker: java.util.concurrent.TimeoutException
	at org.apache.kafka.common.internals.KafkaFutureImpl$SingleWaiter.await(KafkaFutureImpl.java:108)
	at org.apache.kafka.common.internals.KafkaFutureImpl.get(KafkaFutureImpl.java:272)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer.waitForTopicsToBeCreated(KafkaStreamsProducer.java:306)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer.access$000(KafkaStreamsProducer.java:60)
	at io.quarkus.kafka.streams.runtime.KafkaStreamsProducer$1.run(KafkaStreamsProducer.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
```
The same can also happen in `KafkaSnappyProducerITCase`, e.g. [here](https://github.com/quarkusio/quarkus/runs/2309287178?check_suite_focus=true).

There seems to be a configuration issue, because before those timeouts the log is flooded with warnings like these (just en excerpt):
```
[INFO] Running io.quarkus.it.kafka.streams.KafkaStreamsITCase
2021-04-09 22:08:31,378 WARN  [io.qua.deployment] (main) Producing values from constructors and fields is no longer supported and will be removed in a future release: io.quarkus.deployment.annotations.BuildProducer io.quarkus.it.classtransformer.ClassTransformerProcessor.transformers
Executing [/home/runner/work/quarkus/quarkus/integration-tests/kafka-streams/target/quarkus-integration-test-kafka-streams-999-SNAPSHOT-runner, -Dquarkus.http.port=8081, -Dquarkus.http.ssl-port=8444, -Dtest.url=http://localhost:8081, -Dquarkus.log.file.path=/home/runner/work/quarkus/quarkus/integration-tests/kafka-streams/target/quarkus.log, -Dquarkus.log.file.enable=true]
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-04-09 22:08:38,753 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.location' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'sasl.login.refresh.buffer.seconds' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.password' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.truststore.type' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'ssl.endpoint.identification.algorithm' was supplied but isn't a known config.
2021-04-09 22:08:38,779 WARN  [org.apa.kaf.cli.adm.AdminClientConfig] (main) The configuration 'sasl.kerberos.ticket.renew.jitter' was supplied but isn't a known config.
2021-04-09 22:08:38,789 WARN  [org.apa.kaf.str.pro.int.StateDirectory] (main) Using /tmp directory in the state.dir property can cause failures with writing the checkpoint file due to the fact that this directory can be cleared by the OS
```

### PR DESCRIPTION
**Pull request by cescoffier**
Today, we are starting three brokers side-by-side, which include three Kafka instances and three zookeeper instances. That may be too much and may lead to unexpected issues.
This commit split the tests into three different modules:

- kafka verifies the typical Kafka behavior (producer, consumer, codecs...)
- kafka-ssl verifies the connection with SSL
- kafka-sasl verifies the connection with SASL

Each of them starts a single broker with a specific configuration.

In the meantime, I've added a native test for SASL and divided the messaging ITs into 2 groups (one specific to Kafka, and another one for AMQP and others).

### CODE REVIEW

**Comment by famod** [comment]
+💯 for tackling this! I won't be able to review this before this evening and I'm by no means a Kafka expert, so you don't have to wait for my review.

**Comment by famod** [comment]
Btw, shall this fix #16371 XOR #16415 or both?

**Comment by cescoffier** [comment]
Well... actually it does not fixes the issue. I will revisit this.

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by cescoffier** [comment]
@quarkusbot you're drunk... the build succeeded...

@gsmet ^^

**Comment by gsmet** [comment]
It's not the bot that has an issue: the workflow was reported as a failure: https://github.com/quarkusio/quarkus/actions/runs/744900055

Apparently, we have some issue in the workflow file with the native test json

**Comment by gsmet** [comment]
There was an issue in the native tests JSON file. So thanks to the drunk bot for catching it :).

File: integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    private final SaslStrimziKafkaContainer kafka = new SaslStrimziKafkaContainer()
+            .withCopyFileToContainer(MountableFile.forClasspathResource("server.properties"),
+                    "/opt/kafka/config/server.properties");
+
+    @Override
+    public Map<String, String> start() {
+        kafka.start();
+        // Used by the test
+        System.setProperty("bootstrap.servers", kafka.getBootstrapServers());
```
Comment (famod) [inline integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java:30]: Shouldn't the test be able to inject `kafka.bootstrap.servers` (which is added to the map further down) via `@ConfigProperty`? 🤔

**Review by famod — APPROVED** [review]
Phew, a lot to take in and as I said, not a Kafka (or Strimizi, funny name) expert, but LGTM!

I only added a small question.

I haven't had a look at the build logs whether the tests are actually running but I assume you did. 🙂

File: integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java (hunk 1 above)
Comment (cescoffier) [reply integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java:30]: That's actually on purpose to distinguish the value retrieve in the test, and the once injected in the application. You can't inject the configuration value in the (native) test, so I use a system property.

Comment (famod) [reply integration-tests/kafka-sasl/src/test/java/io/quarkus/it/kafka/KafkaSASLTestResource.java:30]: Ah, of course! Thanks.

**Pull request merged**